internal/http.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) MongoDB, Inc. 2022-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package internal // import "go.mongodb.org/mongo-driver/internal"
+
+import (
+	"net"
+	"net/http"
+	"time"
+)
+
+// DefaultHTTPClient is the default HTTP client used across the driver.
+var DefaultHTTPClient = &http.Client{
+	Transport: &http.Transport{
```

Comment:
We should use `http.DefaultTransport.Clone` instead of copying the current `DefaultTransport` config once we change the minimum supported Go version to 1.13. I created [GODRIVER-2623](https://jira.mongodb.org/browse/GODRIVER-2623) to track that improvement.

Add a TODO comment that references [GODRIVER-2623](https://jira.mongodb.org/browse/GODRIVER-2623) and describes the change.

```suggestion
	// TODO(GODRIVER-2623): Use "http.DefaultTransport.Clone" once we change the minimum supported Go version to 1.13.
	Transport: &http.Transport{
```